sdk/core/Azure.Core.TestFramework/README.md

```diff
@@ -402,3 +402,35 @@ To download and unpack all artifacts use the `Download-DevOpsRecordings.ps1` scr
 The `Download-DevOpsRecordings.ps1` would wait for active runs to finish before retrieving artifacts unless `-NoWait` switch is used.
 
 **NOTE:** these scripts require being signed in with Azure CLI (https://docs.microsoft.com/cli/azure/authenticate-azure-cli?view=azure-cli-latest) and access to the internal DevOps project (https://dev.azure.com/azure-sdk/internal/) 
+
+## Note on sharing fields between Generated and Customized code (such as _clientDiagnostics)
+
+Some bindings require code on the customized side to access fields that are generated. For example:
+
+```csharp
+    // Generated\SparkSessionClient.cs
+    public partial class SparkSessionClient
+    {
+        private readonly ClientDiagnostics _clientDiagnostics;
+```
+
+```csharp
+    // Customization\SparkSessionClient.cs
+    internal virtual Response<SparkSession> GetSparkSession(int sessionId, bool? detailed = null, CancellationToken cancellationToken = default)
+    {
+        using var scope = ClientDiagnostics.CreateScope("SparkSessionClient.GetSparkSession");
+```
+
+The direct approach of using _clientDiagnostics here will fail on tests that use the test framework. The proxy generated at runtime uses the constructor with no parameters and _clientDiagnostics will be null.
+
+The correct solution is the use a virtual method, so the testing framework stubs it out:
+
+```csharp
+    internal virtual ClientDiagnostics ClientDiagnostics => _clientDiagnostics;
+```
+
+This may require an InternalsVisisbleTo addition to the test AssemblyInfo.cs:
```

Comment:
```suggestion
Then update your client code to use the new virtual property.  (Pro tip: you can do a VS rename on the field from `_clientDiagnostics` to `ClientDiagnostics` and then change just the field/property definitions back to `_clientDiagnostics` to make this quicker.)

If it's an `internal` property, it will also require an `InternalsVisisbleTo` addition in your AssemblyInfo.cs:
```

You added it to the main AssemblyInfo, not the test one, right?  I thought that was required though it has been awhile since I've done this.